From b573ba9d057617aeb2f2d3429a59b93ef2c4ff3f Mon Sep 17 00:00:00 2001
From: Kelly Ng <ngkelly@google.com>
Date: Tue, 3 Jun 2025 11:54:07 -0700
Subject: [PATCH] Show account selection dialog when a single account exists.

- Recommended fix by security team to prevent auto-importing when there is a single account (see b/388032224#comment4)

FIX: 388032224
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:615666035723ff1828ff24674209012f30082de9)
Merged-In: I83f046c24d45e076aa783e5551b42dcbaacd0ad6
Change-Id: I83f046c24d45e076aa783e5551b42dcbaacd0ad6
---
 .../android/contacts/vcard/SelectAccountActivity.java    | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/com/android/contacts/vcard/SelectAccountActivity.java b/src/com/android/contacts/vcard/SelectAccountActivity.java
index eb13e505c..366a7ead7 100644
--- a/src/com/android/contacts/vcard/SelectAccountActivity.java
+++ b/src/com/android/contacts/vcard/SelectAccountActivity.java
@@ -66,15 +66,6 @@ public class SelectAccountActivity extends Activity {
             Log.w(LOG_TAG, "Account does not exist");
             finish();
             return;
-        } else if (accountList.size() == 1) {
-            final AccountWithDataSet account = accountList.get(0);
-            final Intent intent = new Intent();
-            intent.putExtra(ACCOUNT_NAME, account.name);
-            intent.putExtra(ACCOUNT_TYPE, account.type);
-            intent.putExtra(DATA_SET, account.dataSet);
-            setResult(RESULT_OK, intent);
-            finish();
-            return;
         }
 
         Log.i(LOG_TAG, "The number of available accounts: " + accountList.size());
-- 
2.50.1.351.gcf940e82a1-goog

